src/ir/properties.h

```diff
@@ -240,17 +240,28 @@ inline Index getZeroExtBits(Expression* curr) {
 // way to the final value falling through, potentially through multiple
 // intermediate expressions.
 //
+// The behavior wrt tees is customizable, since in some cases we do not want to
+// look through tees (for example, the type of a tee is related to the local,
+// not the value, so if we returned the fallthrough of the tee we'd have a
+// possible difference between the type in the IR and the type of the value,
+// which some cases care about).
+//
 // TODO: Receive a Module instead of FeatureSet, to pass to EffectAnalyzer?
-inline Expression* getImmediateFallthrough(Expression* curr,
-                                           const PassOptions& passOptions,
-                                           Module& module) {
+
+enum class FallthroughBehavior { AllowTee, NoTee };
```

Comment:
Maybe `AllowTypeNarrowing` and `NoAllowTypeNarrowing` to be more descriptive and to generalize to br_if as well?

Reply:
Hmm, I'm not sure `AllowTypeNarrowing` is quite right. A block can also narrow a type, so the real issue is allowing certain easily-fixable narrowings. A block's narrowing can always be fixed with a refinalize (I think?), but a tee's can't. A tee's can always be fixed in other ways though, like adding a cast, so even "AllowFixableTypeNarrowing" isn't quite right.

Since I can't see a good way to do that, I'd still prefer to have this focus on tee. If we are worried about br_if return values, then separately we could just ignore them in this function (since they are never used in practice, that would be fine).

Reply:
I do think it would be good to handle br_if correctly as well. Presumably we will eventually see fuzzer failures if we don't(?). I can't think of a better name for this property, so how about keeping `AllowTee` and `NoTee`, but also having it apply to br_if?

Reply:
Fair enough, maybe we'll get fuzzer failures some day. Ok, I generalized the name to `AllowTeeBrIf`, a little verbose but it's accurate and the best name I can think of.